src/core/bpf-firewall.c

```diff
@@ -392,29 +426,61 @@ static int bpf_firewall_prepare_access_maps(
                 if (!cc)
                         continue;
 
-                bpf_firewall_count_access_items(verdict == ACCESS_ALLOWED ? cc->ip_address_allow : cc->ip_address_deny, &n_ipv4, &n_ipv6);
+                list = verdict == ACCESS_ALLOWED ? cc->ip_address_allow : cc->ip_address_deny;
+
+                bpf_firewall_count_access_items(list, &n_ipv4, &n_ipv6);
+
+                if (!LIST_IS_EMPTY(list) && n_ipv4 == 1 && n_ipv6 == 1 && !ipv4_any && !ipv6_any)
+                        // There should be corresponding IPv4 and IPv6 entries in each list so
+                        // checking just the first item is okay
```

Comment:
we generally use `//` comments only for temporary "FIXME" style comments, not for explanatory comments that should stay in the code forever. See the second rule:

https://systemd.io/CODING_STYLE#formatting